docs/modules/ROOT/pages/documentation/payara-server/server-configuration/certificate-management.adoc

```diff
@@ -0,0 +1,490 @@
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--distinguishedname`, `--dn`
+|String
+|The distinguished name to use when generating the certificate.
+|N/A
+|yes
+
+|`--alternativenames`, `--altnames`
+|String[]
+|The semi-colon (;) separated list of additional Subject Alternative Names to add to the generated certificate.
+|N/A
+|no
+
+|`--alias` (primary)
+|String
+|The alias name to use when generating the certificate and storing it in the key and trust stores.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> generate-self-signed-certificate --dn "CN=test.payara.fish,IP=192.168.1.1"
+--listener http-listener-2
+--alternativenames "test2.payara.fish;DNS:test3.payara.fish,IP:127.0.0.1,EMAIL:anon@payara.fish"
+--target Instance1
+test_cert
+----
+
+[[generate-csr]]
+== `generate-csr`
+
+*Usage*::
+`asadmin> generate-csr --target=instancename --listener=listenername alias`
+
+*Aim*::
+This command can generate a certificate signing request (CSR) for an instance or listener's self-signed certificate,
+placing the resultant CSR file in ${installDir}/glassfish/tls, using the alias name as the file name.
+
+*Note::
+This will overwrite a CSR with the same name already present in the ${installDir}/glassfish/tls directory.
+
+[[command-options]]
+=== Command Options
+
+[cols="3,1,5,1,1",options="header"]
+|===
+|Option
+|Type
+|Description
+|Default
+|Mandatory
+
+|`--target`
+|String
+|The name of the instance to get the certificate from.
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to get the certificate from.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--alias` (primary)
+|String
+|The alias name of the certificate to generate a CSR for.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> generate-csr --listener http-listener-2 --target Instance1 test_cert
+----
+
+[[add-to-keystore]]
+== `add-to-keystore`
+
+*Usage*::
+`asadmin> add-to-keystore --target=instancename --listener=listenername --file /path/to/file alias`
+
+*Aim*::
+This command adds a certificate bundle (e.g. .p12 or .jks file) or certificate (e.g. .cert file) to the target instance
+or listener's key store using the provided alias.
+
+If the instance or listener is configured to use the default key store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been added to the default key store of
+the DAS), since any certificates added to the instance stores would be lost upon next synchronisation.
```

Comment:
Again the same simplification I suggested above:
```suggestion
If the instance or listener is configured to use the default key store, the command will place the certificate bundle in the default key store of the DAS and then synchronise
the instance with the DAS. This is to preserve the certificates across instance restarts because any certificates added directly to the instance stores would be lost upon next synchronisation.
```

Reply:
See comments elsewhere, your suggestion isn't what happens.